From ca581e3e0d6565133de3b2dcd8aa81679d462dcb Mon Sep 17 00:00:00 2001
From: jomini <hexh0628@thundersoft.com>
Date: Wed, 9 Apr 2014 15:55:01 +0800
Subject: [PATCH] ARM:dts:msm: add cpt claa053wd41_3xb devicetree

Devicetree support lcd truly cpt claa053wd41_3xb ili9881 driver

Change-Id: I90848a7140ba2792d5204b6dae8320bb81ad75cf
Signed-off-by: Xiaohui He <hexh0628@thundersoft.com>
---
 ...nel-cpt-claa053wd41-3xb-ili9881-720p-video.dtsi |  221 ++++++++++++++++
 arch/arm/boot/dts/msm8226-mdss-panels.dtsi         |    3 +-
 arch/arm/boot/dts/msm8226-v2-qrd-dvt.dts           |    6 +-
 ...anel_cpt_claa053wd41_3xb_ili9881_720p_video.xml |  278 ++++++++++++++++++++
 4 files changed, 504 insertions(+), 4 deletions(-)
 create mode 100644 arch/arm/boot/dts/dsi-panel-cpt-claa053wd41-3xb-ili9881-720p-video.dtsi
 create mode 100644 arch/arm/boot/dts/panel_cpt_claa053wd41_3xb_ili9881_720p_video.xml

diff --git a/arch/arm/boot/dts/dsi-panel-cpt-claa053wd41-3xb-ili9881-720p-video.dtsi b/arch/arm/boot/dts/dsi-panel-cpt-claa053wd41-3xb-ili9881-720p-video.dtsi
new file mode 100644
index 0000000..959f4b8
--- /dev/null
+++ b/arch/arm/boot/dts/dsi-panel-cpt-claa053wd41-3xb-ili9881-720p-video.dtsi
@@ -0,0 +1,221 @@
+/* Copyright (c) 2014, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/*---------------------------------------------------------------------------
+ * This file is autogenerated file using gcdb parser. Please do not edit it.
+ * Update input XML file to add a new entry or update variable in this file
+ * VERSION = "1.0"
+ *---------------------------------------------------------------------------*/
+&mdss_mdp {
+	dsi_cpt_claa053wd41_3xb_ili9881_720p_video: qcom,mdss_dsi_cpt_claa053wd41_3xb_ili9881_720p_video {
+		qcom,mdss-dsi-panel-name = "ili9881 720p video mode dsi panel";
+		qcom,mdss-dsi-panel-controller = <&mdss_dsi0>;
+		qcom,mdss-dsi-panel-type = "dsi_video_mode";
+		qcom,mdss-dsi-panel-destination = "display_1";
+		qcom,mdss-dsi-panel-framerate = <60>;
+		qcom,mdss-dsi-virtual-channel-id = <0>;
+		qcom,mdss-dsi-stream = <0>;
+		qcom,mdss-dsi-panel-width = <720>;
+		qcom,mdss-dsi-panel-height = <1280>;
+		qcom,mdss-dsi-h-front-porch = <90>;
+		qcom,mdss-dsi-h-back-porch = <90>;
+		qcom,mdss-dsi-h-pulse-width = <6>;
+		qcom,mdss-dsi-h-sync-skew = <0>;
+		qcom,mdss-dsi-v-back-porch = <10>;
+		qcom,mdss-dsi-v-front-porch = <6>;
+		qcom,mdss-dsi-v-pulse-width = <4>;
+		qcom,mdss-dsi-h-left-border = <0>;
+		qcom,mdss-dsi-h-right-border = <0>;
+		qcom,mdss-dsi-v-top-border = <0>;
+		qcom,mdss-dsi-v-bottom-border = <0>;
+		qcom,mdss-dsi-bpp = <24>;
+		qcom,mdss-dsi-color-order = <0>;
+		qcom,mdss-dsi-underflow-color = <0xff>;
+		qcom,mdss-dsi-border-color = <0>;
+		qcom,mdss-dsi-on-command = [39 01 00 00 00 00 04 FF 98 81 07
+			39 01 00 00 00 00 02 03 20
+			39 01 00 00 00 00 02 04 06
+			39 01 00 00 00 00 02 05 00
+			39 01 00 00 00 00 02 06 00
+			39 01 00 00 00 00 02 07 00
+			39 01 00 00 00 00 02 08 00
+			39 01 00 00 00 00 02 09 00
+			39 01 00 00 00 00 02 0a 01
+			39 01 00 00 00 00 02 0b 01
+			39 01 00 00 00 00 02 0c 00
+			39 01 00 00 00 00 02 0d 00
+			39 01 00 00 00 00 02 0e 00
+			39 01 00 00 00 00 02 0f 00
+			39 01 00 00 00 00 02 10 40
+			39 01 00 00 00 00 02 11 02
+			39 01 00 00 00 00 02 12 05
+			39 01 00 00 00 00 02 13 00
+			39 01 00 00 00 00 02 14 00
+			39 01 00 00 00 00 02 15 00
+			39 01 00 00 00 00 02 16 01
+			39 01 00 00 00 00 02 17 01
+			39 01 00 00 00 00 02 18 00
+			39 01 00 00 00 00 02 19 00
+			39 01 00 00 00 00 02 1a 00
+			39 01 00 00 00 00 02 1b 50
+			39 01 00 00 00 00 02 1c BB
+			39 01 00 00 00 00 02 1d 0C
+			39 01 00 00 00 00 02 1e 00
+			39 01 00 00 00 00 02 1f 00
+			39 01 00 00 00 00 02 20 00
+			39 01 00 00 00 00 02 21 00
+			39 01 00 00 00 00 02 22 00
+			39 01 00 00 00 00 02 23 C0
+			39 01 00 00 00 00 02 24 30
+			39 01 00 00 00 00 02 25 00
+			39 01 00 00 00 00 02 26 00
+			39 01 00 00 00 00 02 27 03
+			39 01 00 00 00 00 02 30 01
+			39 01 00 00 00 00 02 31 23
+			39 01 00 00 00 00 02 32 45
+			39 01 00 00 00 00 02 33 67
+			39 01 00 00 00 00 02 34 89
+			39 01 00 00 00 00 02 35 ab
+			39 01 00 00 00 00 02 36 01
+			39 01 00 00 00 00 02 37 23
+			39 01 00 00 00 00 02 38 45
+			39 01 00 00 00 00 02 39 67
+			39 01 00 00 00 00 02 3a 89
+			39 01 00 00 00 00 02 3b ab
+			39 01 00 00 00 00 02 3c cd
+			39 01 00 00 00 00 02 3d ef
+			39 01 00 00 00 00 02 50 11
+			39 01 00 00 00 00 02 51 06
+			39 01 00 00 00 00 02 52 02
+			39 01 00 00 00 00 02 53 0c
+			39 01 00 00 00 00 02 54 0e
+			39 01 00 00 00 00 02 55 0d
+			39 01 00 00 00 00 02 56 0f
+			39 01 00 00 00 00 02 57 05
+			39 01 00 00 00 00 02 58 02
+			39 01 00 00 00 00 02 59 02
+			39 01 00 00 00 00 02 5a 02
+			39 01 00 00 00 00 02 5b 02
+			39 01 00 00 00 00 02 5c 02
+			39 01 00 00 00 00 02 5d 02
+			39 01 00 00 00 00 02 5e 02
+			39 01 00 00 00 00 02 5f 02
+			39 01 00 00 00 00 02 60 02
+			39 01 00 00 00 00 02 61 02
+			39 01 00 00 00 00 02 62 02
+			39 01 00 00 00 00 02 63 08
+			39 01 00 00 00 00 02 64 02
+			39 01 00 00 00 00 02 65 01
+			39 01 00 00 00 00 02 66 00
+			39 01 00 00 00 00 02 67 08
+			39 01 00 00 00 00 02 68 02
+			39 01 00 00 00 00 02 69 0c
+			39 01 00 00 00 00 02 6a 0e
+			39 01 00 00 00 00 02 6b 0d
+			39 01 00 00 00 00 02 6c 0f
+			39 01 00 00 00 00 02 6d 05
+			39 01 00 00 00 00 02 6e 02
+			39 01 00 00 00 00 02 6f 02
+			39 01 00 00 00 00 02 70 02
+			39 01 00 00 00 00 02 71 02
+			39 01 00 00 00 00 02 72 02
+			39 01 00 00 00 00 02 73 02
+			39 01 00 00 00 00 02 74 02
+			39 01 00 00 00 00 02 75 02
+			39 01 00 00 00 00 02 76 02
+			39 01 00 00 00 00 02 77 02
+			39 01 00 00 00 00 02 78 02
+			39 01 00 00 00 00 02 79 06
+			39 01 00 00 00 00 02 7a 02
+			39 01 00 00 00 00 02 7b 01
+			39 01 00 00 00 00 02 7c 00
+			39 01 00 00 00 00 04 FF 98 81 08
+			39 01 00 00 00 00 02 76 E4
+			39 01 00 00 00 00 02 78 0C
+			39 01 00 00 00 00 02 74 2A
+			39 01 00 00 00 00 02 8E 1A
+			39 01 00 00 00 00 02 98 BF
+			39 01 00 00 00 00 02 DF 01
+			39 01 00 00 00 00 02 E3 00
+			39 01 00 00 00 00 02 97 08
+			39 01 00 00 00 00 04 FF 98 81 01
+			39 01 00 00 00 00 02 22 0A
+			39 01 00 00 00 00 02 53 96
+			39 01 00 00 00 00 02 50 80
+			39 01 00 00 00 00 02 51 80
+			39 01 00 00 00 00 02 A0 00
+			39 01 00 00 00 00 02 A1 0F
+			39 01 00 00 00 00 02 A2 1C
+			39 01 00 00 00 00 02 A3 12
+			39 01 00 00 00 00 02 A4 13
+			39 01 00 00 00 00 02 A5 26
+			39 01 00 00 00 00 02 A6 19
+			39 01 00 00 00 00 02 A7 1A
+			39 01 00 00 00 00 02 A8 7E
+			39 01 00 00 00 00 02 A9 19
+			39 01 00 00 00 00 02 AA 22
+			39 01 00 00 00 00 02 AB 85
+			39 01 00 00 00 00 02 AC 1B
+			39 01 00 00 00 00 02 AD 1E
+			39 01 00 00 00 00 02 AE 56
+			39 01 00 00 00 00 02 AF 2D
+			39 01 00 00 00 00 02 B0 2C
+			39 01 00 00 00 00 02 B1 63
+			39 01 00 00 00 00 02 B2 72
+			39 01 00 00 00 00 02 B3 39
+			39 01 00 00 00 00 02 C0 00
+			39 01 00 00 00 00 02 C1 0F
+			39 01 00 00 00 00 02 C2 1B
+			39 01 00 00 00 00 02 C3 12
+			39 01 00 00 00 00 02 C4 13
+			39 01 00 00 00 00 02 C5 25
+			39 01 00 00 00 00 02 C6 1A
+			39 01 00 00 00 00 02 C7 1D
+			39 01 00 00 00 00 02 C8 7D
+			39 01 00 00 00 00 02 C9 1E
+			39 01 00 00 00 00 02 CA 2D
+			39 01 00 00 00 00 02 CB 81
+			39 01 00 00 00 00 02 CC 21
+			39 01 00 00 00 00 02 CD 1F
+			39 01 00 00 00 00 02 CE 53
+			39 01 00 00 00 00 02 CF 23
+			39 01 00 00 00 00 02 D0 28
+			39 01 00 00 00 00 02 D1 64
+			39 01 00 00 00 00 02 D2 72
+			39 01 00 00 00 00 02 D3 39
+			39 01 00 00 00 00 04 FF 98 81 00
+			05 01 00 00 96 00 02 11 00
+			05 01 00 00 00 00 02 29 00];
+		qcom,mdss-dsi-off-command = [05 01 00 00 32 00 02 28 00
+					05 01 00 00 78 00 02 10 00];
+		qcom,mdss-dsi-on-command-state = "dsi_lp_mode";
+		qcom,mdss-dsi-off-command-state = "dsi_hs_mode";
+		qcom,mdss-dsi-h-sync-pulse = <1>;
+		qcom,mdss-dsi-traffic-mode = <2>;
+		qcom,mdss-dsi-lane-map = <0>;
+		qcom,mdss-dsi-bllp-eof-power-mode;
+		qcom,mdss-dsi-bllp-power-mode;
+		qcom,mdss-dsi-lane-0-state;
+		qcom,mdss-dsi-lane-1-state;
+		qcom,mdss-dsi-lane-2-state;
+		qcom,mdss-dsi-lane-3-state;
+		qcom,mdss-dsi-panel-timings = [7A 1A 12 00 3E 42 16 1E 14 03 04 00];
+		qcom,mdss-dsi-t-clk-post = <0x1f>;
+		qcom,mdss-dsi-t-clk-pre = <0x2d>;
+		qcom,mdss-dsi-bl-min-level = <1>;
+		qcom,mdss-dsi-bl-max-level = <4095>;
+		qcom,mdss-dsi-dma-trigger = <4>;
+		qcom,mdss-dsi-mdp-trigger = <0>;
+		qcom,mdss-dsi-bl-pmic-control-type = "bl_ctrl_wled";
+		qcom,mdss-dsi-reset-sequence = <1 20>, <0 1>, <1 20>;
+	};
+};
diff --git a/arch/arm/boot/dts/msm8226-mdss-panels.dtsi b/arch/arm/boot/dts/msm8226-mdss-panels.dtsi
index 0731a9a..6f3c47a 100644
--- a/arch/arm/boot/dts/msm8226-mdss-panels.dtsi
+++ b/arch/arm/boot/dts/msm8226-mdss-panels.dtsi
@@ -1,4 +1,4 @@
-/* Copyright (c) 2013, The Linux Foundation. All rights reserved.
+/* Copyright (c) 2013-2014, The Linux Foundation. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 and
@@ -17,3 +17,4 @@
 /include/ "dsi-panel-nt35590-720p-cmd.dtsi"
 /include/ "dsi-panel-ssd2080m-720p-video.dtsi"
 /include/ "dsi-panel-jdi-1080p-video.dtsi"
+/include/ "dsi-panel-cpt-claa053wd41-3xb-ili9881-720p-video.dtsi"
diff --git a/arch/arm/boot/dts/msm8226-v2-qrd-dvt.dts b/arch/arm/boot/dts/msm8226-v2-qrd-dvt.dts
index 13402af..e93da76 100644
--- a/arch/arm/boot/dts/msm8226-v2-qrd-dvt.dts
+++ b/arch/arm/boot/dts/msm8226-v2-qrd-dvt.dts
@@ -1,4 +1,4 @@
-/* Copyright (c) 2013, The Linux Foundation. All rights reserved.
+/* Copyright (c) 2013-2014, The Linux Foundation. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 and
@@ -21,10 +21,10 @@
 };
 
 &mdss_dsi0 {
-	qcom,dsi-pref-prim-pan = <&dsi_hx8394a_720_vid>;
+	qcom,dsi-pref-prim-pan = <&dsi_cpt_claa053wd41_3xb_ili9881_720p_video>;
 };
 
-&dsi_hx8394a_720_vid {
+&dsi_cpt_claa053wd41_3xb_ili9881_720p_video {
 	qcom,cont-splash-enabled;
 };
 
diff --git a/arch/arm/boot/dts/panel_cpt_claa053wd41_3xb_ili9881_720p_video.xml b/arch/arm/boot/dts/panel_cpt_claa053wd41_3xb_ili9881_720p_video.xml
new file mode 100644
index 0000000..28d3d0c
--- /dev/null
+++ b/arch/arm/boot/dts/panel_cpt_claa053wd41_3xb_ili9881_720p_video.xml
@@ -0,0 +1,278 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+# Copyright (c) 2014, The Linux Foundation. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#     * Neither the name of The Linux Foundation nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED
+# WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+-->
+
+<GCDB>
+  <Version>"1.0"</Version>
+  <PanelId>cpt-claa053wd41-3xb-ili9881-720p-video</PanelId>
+  <PanelH>cpt_claa053wd41_3xb_ili9881_720p_video</PanelH>
+  <PanelEntry>
+
+	<!-- Panel configuration -->
+	<PanelName>"ili9881 720p video mode dsi panel"</PanelName>
+	<PanelController>"mdss_dsi0"</PanelController>
+	<PanelInterface>10</PanelInterface>
+	<PanelType>0</PanelType>
+	<PanelDestination>"DISPLAY_1"</PanelDestination>
+	<PanelOrientation>0</PanelOrientation>
+	<PanelFrameRate>60</PanelFrameRate>
+	<PanelChannelId>0</PanelChannelId>
+	<DSIVirtualChannelId>0</DSIVirtualChannelId>
+	<PanelBroadcastMode>0</PanelBroadcastMode>
+	<!-- Optional Panel configuration -->
+	<!--BitClockFrequency>0</BitClockFrequency -->
+	<DSIStream>0</DSIStream>
+	<PanelCompatible>"qcom,mdss-dsi-panel"</PanelCompatible>
+	<InterleaveMode>0</InterleaveMode>
+
+	<!-- Panel Resolution -->
+	<PanelWidth>720</PanelWidth>
+	<PanelHeight>1280</PanelHeight>
+	<HFrontPorch>90</HFrontPorch>
+	<HBackPorch>90</HBackPorch>
+	<HPulseWidth>6</HPulseWidth>
+	<HSyncSkew>0</HSyncSkew>
+	<VBackPorch>10</VBackPorch>
+	<VFrontPorch>6</VFrontPorch>
+	<VPulseWidth>4</VPulseWidth>
+	<HLeftBorder>0</HLeftBorder>
+	<HRightBorder>0</HRightBorder>
+	<VTopBorder>0</VTopBorder>
+	<VBottomBorder>0</VBottomBorder>
+	<!-- Optional Panel resolution configuration -->
+	<!--HActiveRes>0</HActiveRes>
+	<VActiveRes>100</VActiveRes>
+	<InvertDataPolarity>0</InvertDataPolarity>
+	<InvertVsyncPolarity>0</InvertVsyncPolarity>
+	<InvertHsyncPolarity>0</InvertHsyncPolarity -->
+
+	<!-- Panel Color Information -->
+	<ColorFormat>24</ColorFormat>
+	<ColorOrder>0</ColorOrder>
+	<UnderFlowColor>0xff</UnderFlowColor>
+	<BorderColor>0</BorderColor>
+
+	<!-- Panel Command information -->
+	<OnCommand>"0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x04, 0xFF, 0x98, 0x81, 0x07,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x03, 0x20,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x04, 0x06,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x05, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x06, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x07, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x08, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x09, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0a, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0b, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0c, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0d, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0e, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x0f, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x10, 0x40,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x11, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x12, 0x05,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x13, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x14, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x15, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x16, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x17, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x18, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x19, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x1a, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x1b, 0x50,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x1c, 0xBB,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x1d, 0x0C,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x1e, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x1f, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x20, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x21, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x22, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x23, 0xC0,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x24, 0x30,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x25, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x26, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x27, 0x03,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x30, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x31, 0x23,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x32, 0x45,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x33, 0x67,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x34, 0x89,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x35, 0xab,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x36, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x37, 0x23,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x38, 0x45,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x39, 0x67,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3a, 0x89,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3b, 0xab,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3c, 0xcd,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x3d, 0xef,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x50, 0x11,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x51, 0x06,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x52, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x53, 0x0c,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x54, 0x0e,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x55, 0x0d,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x56, 0x0f,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x57, 0x05,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x58, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x59, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x5a, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x5b, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x5c, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x5d, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x5e, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x5f, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x60, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x61, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x62, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x63, 0x08,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x64, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x65, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x66, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x67, 0x08,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x68, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x69, 0x0c,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x6a, 0x0e,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x6b, 0x0d,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x6c, 0x0f,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x6d, 0x05,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x6e, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x6f, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x70, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x71, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x72, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x73, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x74, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x75, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x76, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x77, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x78, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x79, 0x06,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x7a, 0x02,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x7b, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x7c, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x04, 0xFF, 0x98, 0x81, 0x08,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x76, 0xE4,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x78, 0x0C,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x74, 0x2A,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x8E, 0x1A,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x98, 0xBF,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xDF, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xE3, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x97, 0x08,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x04, 0xFF, 0x98, 0x81, 0x01,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x22, 0x0A,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x53, 0x96,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x50, 0x80,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x51, 0x80,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA0, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA1, 0x0F,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA2, 0x1C,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA3, 0x12,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA4, 0x13,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA5, 0x26,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA6, 0x19,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA7, 0x1A,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA8, 0x7E,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xA9, 0x19,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xAA, 0x22,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xAB, 0x85,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xAC, 0x1B,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xAD, 0x1E,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xAE, 0x56,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xAF, 0x2D,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xB0, 0x2C,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xB1, 0x63,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xB2, 0x72,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xB3, 0x39,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC0, 0x00,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC1, 0x0F,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC2, 0x1B,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC3, 0x12,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC4, 0x13,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC5, 0x25,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC6, 0x1A,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC7, 0x1D,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC8, 0x7D,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xC9, 0x1E,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xCA, 0x2D,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xCB, 0x81,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xCC, 0x21,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xCD, 0x1F,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xCE, 0x53,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xCF, 0x23,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xD0, 0x28,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xD1, 0x64,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xD2, 0x72,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0xD3, 0x39,
+			0x39, 0x01, 0x00, 0x00, 0x00, 0x00, 0x04, 0xFF, 0x98, 0x81, 0x00,
+			0x05, 0x01, 0x00, 0x00, 0x96, 0x00, 0x02, 0x11, 0x00,
+			0x05, 0x01, 0x00, 0x00, 0x00, 0x00, 0x02, 0x29, 0x00"</OnCommand>
+	<OffCommand>"0x05, 0x01, 0x00, 0x00, 0x32, 0x00, 0x02, 0x28, 0x00,
+					0x05, 0x01, 0x00, 0x00, 0x78, 0x00, 0x02, 0x10, 0x00"</OffCommand>
+	<OnCommandState>0</OnCommandState>
+	<OffCommandState>1</OffCommandState>
+
+	<!-- Video mode panel information -->
+	<HSyncPulse>1</HSyncPulse>
+	<HFPPowerMode>0</HFPPowerMode>
+	<HBPPowerMode>0</HBPPowerMode>
+	<HSAPowerMode>0</HSAPowerMode>
+	<BLLPEOFPowerMode>1</BLLPEOFPowerMode>
+	<BLLPPowerMode>1</BLLPPowerMode>
+	<TrafficMode>2</TrafficMode>
+	<DMADelayAfterVsync>0</DMADelayAfterVsync>
+	<BLLPEOFPower>0x9</BLLPEOFPower>
+
+	<!-- Lane Configuration -->
+	<DSILanes>4</DSILanes>
+	<DSILaneMap>0</DSILaneMap>
+	<Lane0State>1</Lane0State>
+	<Lane1State>1</Lane1State>
+	<Lane2State>1</Lane2State>
+	<Lane3State>1</Lane3State>
+
+	<!-- Panel Timing -->
+	<PanelTimings>"0x7A, 0x1A, 0x12, 0x00, 0x3E, 0x42, 0x16, 0x1E, 0x14, 0x03, 0x04, 0x00"</PanelTimings>
+	<DSIMDPTrigger>0</DSIMDPTrigger>
+	<DSIDMATrigger>4</DSIDMATrigger>
+	<TClkPost>0x1f</TClkPost>
+	<TClkPre>0x2d</TClkPre>
+
+	<!-- Backlight -->
+	<BLInterfaceType>1</BLInterfaceType>
+	<BLMinLevel>1</BLMinLevel>
+	<BLMaxLevel>4095</BLMaxLevel>
+	<BLStep>100</BLStep>
+	<BLPMICModel>"PMIC_8941"</BLPMICModel>
+	<BLPMICControlType>1</BLPMICControlType>
+
+  </PanelEntry>
+</GCDB>
+
-- 
1.7.4.1

